src/python/pants/base/exception_sink.py

```diff
@@ -433,6 +433,18 @@ def _handle_signal_gracefully(cls, signum, signame, traceback_lines):
     # Exit, printing the output to the terminal.
     cls._exit_with_failure(terminal_log_entry)
 
+  @classmethod
+  @contextmanager
+  def exiter_as(cls, exiter):
+    """A contextmanager which temporarily overrides the exiter."""
+    try:
+      previous_exiter = cls._exiter if cls._exiter else exiter
+      cls._exiter = exiter
```

Comment:
I don't think this is needed, as `ExceptionSink.reset_exiter()` already overrides the class variables :)

Reply:
We might need to edit `reset_exiter()` the way we did `reset_signal_handler()` in order to support `trapped_signals()` in #6574. Luckily all that would mean is making `reset_exiter()` return the previous exiter.

https://github.com/pantsbuild/pants/blob/f9b0313d11a1a386abcd97e5f15cf3eb0ff14168/src/python/pants/base/exception_sink.py#L278-L311